src/utils/api.js

```diff
@@ -13,3 +13,10 @@ export const getArticles = () => {
         return articleCardProps;
     })
 }
+
+export const getSingleArticle = (articleId) => {
+    return articlesApi.get(`/articles/${articleId}`).then((res) => {
+        return res.data;
```

Comment:
i would return res.data.article here to avoid having to use [0] when setting your state